src/ArborX_LinearBVH.hpp

```diff
@@ -144,9 +144,10 @@ BasicBoundingVolumeHierarchy<MemorySpace, BoundingVolume, Enable>::
   static_assert(
       KokkosExt::is_accessible_from<MemorySpace, ExecutionSpace>::value);
   Details::check_valid_access_traits(PrimitivesTag{}, primitives);
-  using Access = AccessTraits<Primitives, PrimitivesTag>;
-  static_assert(KokkosExt::is_accessible_from<typename Access::memory_space,
-                                              ExecutionSpace>::value,
+  Details::Adapt adapt_primitives(PrimitivesTag(), primitives);
```

Comment:
I'd rather use `primitives` internally, and `user_primitives` in the interface functions. Same with predicates.